numba/targets/linalg.py

```diff
@@ -72,6 +72,201 @@ def make_constant_slot(context, builder, ty, val):
+                         types.intp              # lda
+                         )
+        return types.ExternalFunction("numba_xxpotrf", sig)
+
+    @classmethod
+    def numba_ez_gesdd(cls, dtype):
+        stype = getattr(dtype, "underlying_float", dtype)
+        sig = types.intc(
+            types.char,             # kind
+            types.char,             # jobz
+            types.intp,             # m
+            types.intp,             # n
+            types.CPointer(dtype),  # a
+            types.intp,             # lda
+            types.CPointer(stype),  # s
+            types.CPointer(dtype),  # u
+            types.intp,             # ldu
+            types.CPointer(dtype),  # vt
+            types.intp              # ldvt
+        )
+
+        return types.ExternalFunction("numba_ez_gesdd", sig)
+
+    @classmethod
+    def numba_ez_geqrf(cls, dtype):
+        sig = types.intc(
+            types.char,             # kind
+            types.intp,             # m
+            types.intp,             # n
+            types.CPointer(dtype),  # a
+            types.intp,             # lda
+            types.CPointer(dtype),  # tau
+        )
+        return types.ExternalFunction("numba_ez_geqrf", sig)
+
+    @classmethod
+    def numba_ez_xxgqr(cls, dtype):
+        sig = types.intc(
+            types.char,             # kind
+            types.intp,             # m
+            types.intp,             # n
+            types.intp,             # k
+            types.CPointer(dtype),  # a
+            types.intp,             # lda
+            types.CPointer(dtype),  # tau
+        )
+        return types.ExternalFunction("numba_ez_xxgqr", sig)
+
+    @classmethod
+    def numba_ez_gelsd(cls, dtype):
+        rtype = getattr(dtype, "underlying_float", dtype)
+        sig = types.intc(
+            types.char,                 # kind
+            types.intp,                 # m
+            types.intp,                 # n
+            types.intp,                 # nrhs
+            types.CPointer(dtype),      # a
+            types.intp,                 # lda
+            types.CPointer(dtype),      # b
+            types.intp,                 # ldb
+            types.CPointer(rtype),      # S
+            types.float64,              # rcond
+            types.CPointer(types.intc)  # rank
+        )
+        return types.ExternalFunction("numba_ez_gelsd", sig)
+
+    @classmethod
+    def numba_xgesv(cls, dtype):
+        sig = types.intc(
+            types.char,                    # kind
+            types.intp,                    # n
+            types.intp,                    # nhrs
+            types.CPointer(dtype),         # a
+            types.intp,                    # lda
+            types.CPointer(F_INT_nbtype),  # ipiv
+            types.CPointer(dtype),         # b
+            types.intp                     # ldb
+        )
+        return types.ExternalFunction("numba_xgesv", sig)
+
+
+lapack = _LAPACK()
```

Comment:
You may have missed the previous comment, but these shouldn't be initialized at module toplevel since it will fail if e.g. Scipy is not present (it will also slow down compiler initialization). Just instantiate them where they are used, i.e. in the various implementations.